easybuild/tools/modules.py

```diff
@@ -681,13 +681,42 @@ def modpath_extensions_for(self, mod_names):
         # copy environment so we can restore it
         env = os.environ.copy()
 
+        # regex for $MODULEPATH definitions in Python syntax, e.g.:
+        # os.environ['MODULEPATH'] = "/tmp/example:/home/example/modules:";
+        modpath_ext_regex = re.compile(r"^os\.environ\[.MODULEPATH.\]\s*=\s*(?:'|\")(\S+)(?:'|\");?", re.M)
+
         modpath_exts = {}
         for mod_name in mod_names:
-            modtxt = self.read_module_file(mod_name)
-            useregex = re.compile(r'^\s*module\s+use\s+"?([^"\s]+)"?', re.M)
-            exts = useregex.findall(modtxt)
 
-            self.log.debug("Found $MODULEPATH extensions for %s: %s" % (mod_name, exts))
+            # module must be available to (fake) load
+            if not self.exist([mod_name])[0]:
+                raise EasyBuildError("Can not determine $MODULEPATH extensions for non-existing module %s", mod_name)
+
+            # grab current value for $MODULEPATH, so we can figure out which paths get added to it
+            curr_modulepath = [p for p in os.environ.get('MODULEPATH', '').split(os.pathsep) if p]
+
+            # fake load for module: just obtain changes that would be made to environment, don't apply them (yet)
+            # this is prefrred over parsing the module file contents, to take into account conditional extensions
+            mod_load_txt = self.run_module('load', mod_name, return_output=True)
```

Comment:
I switched to a 'fake load' rather than using `module show` (as was done in #1474) to dance around the fact that `modulecmd.tcl` is not showing changes to `$MODULEPATH` if the module file includes `module use` statements, although it *does* properly extend `$MODULEPATH` when the module is loaded...

This approach was taken over just 'parsing' the module file itself because some of the `module use` statements may be conditional, i..e when `--subdir-user-modules` is used (cfr. `make_module_extend_modpath`); also, it makes parsing easier, since the output is the same regardless of Tcl or Lua module files.

However, neither the `show` nor the 'fake load' approach work, because they're too dependent on the environment in which they're being run...

It's possible that the `$MODULEPATH` extensions performed by a module were already done in one way or another in the current environment, which will cause us to miss extensions; resetting `$MODULEPATH` is not going to work out, since then we may break the `load` of not only the module being inspected, but also of `load` statements included in that module...

Also, strictly speaking this implementation gives wrong results for `icc`/`ifort` (cfr. new `modpath_extensions_for` test), since it'll result in an empty list of `$MODULEPATH` extensions for the `ifort` module, because the `icc` module that contains the same extensions will be loaded by the time the `ifort` module is inspected.

I need to think about a proper way to deal with all of this... It seems like we'll need to go back to parsing the actual module file, and somehow figure out ourselves whether the `use` statements are conditional... Or, maybe we don't need to bother: does it matter whether or not extensions that may not actually happen are included?

@geimer: thoughts?